app/src/main/java/com/vultisig/wallet/ui/models/VaultAccountsViewModel.kt

```diff
@@ -105,6 +117,19 @@ internal class VaultAccountsViewModel @Inject constructor(
             }
             accountsRepository
                 .loadAddresses(vaultId)
+                .onEach { addresses->
+                    addresses.forEach{
+                        val addressOrder = chainsOrderRepository.find(it.chain.raw)
+                        if (addressOrder == null)
+                            chainsOrderRepository.insert(it.chain.raw)
+                    }
+                }
+                .zip(chainsOrderRepository.loadByOrders()) { addresses, chainOrders ->
+                    chainOrders.map {orderEntity->
+                         addresses.find { address -> address.chain.raw == orderEntity.value }!!
+                    }
+                }
```

Comment:
That's suboptimal. Try to do all these mappings in one cycle. I suggest putting chainOrders in a Map. And making sure there's no !! anywhere.

Reply:
Thanks, I used your suggested approach and updated the code